### ISSUE TITLE
Buildpack fails when Gemfile defines Ruby version programmatically

### ISSUE THREAD

**Issue by lnhrdt** [cloudfoundry/ruby-buildpack#61]
In my RoR projects, I define the Ruby version in my Gemfile like so:

```ruby
source 'https://rubygems.org'
ruby File.read('.ruby-version').strip
...
```

so that the Ruby version is synchronized with with rbenv (which I use in local development). When I try to deploy my application using using `cloudfoundry/ruby-buildpack` (PWS/v1.6.19), the buildpack cannot parse the Gemfile and fails (log included below).

I'd expect `cloudfoundry/ruby-buildpack` to configure the Ruby version just as if I had written it like so:

```ruby
source 'https://rubygems.org'
ruby '2.3.1'
...
```

FWIW, my Gemfile works fine locally and using `heroku/heroku-buildpack-ruby` on Heroku.

**Log:**

```
Staging...
-------> Buildpack version 1.6.19
       Downloaded [file:///tmp/buildpacks/fda2b356ce7de642197a27faa3892370/dependencies/https___pivotal-buildpacks.s3.amazonaws.com_concourse-binaries_bundler_bundler-1.12.5.tgz]
-----> Compiling Ruby/Rails
sh: 1: Syntax error: EOF in backquote substitution
./Gemfile:2:in `ruby_requirement': undefined method `read' for nil:NilClass (NoMethodError)
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_semver_version.rb:32:in `instance_eval'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_semver_version.rb:32:in `ruby_requirement'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_semver_version.rb:7:in `initialize'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_version.rb:126:in `new'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_version.rb:38:in `initialize'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby_version.rb:126:in `update_version'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:187:in `block in ruby_version'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:180:in `ruby_version'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:331:in `block in install_ruby'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:187:in `new'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:330:in `install_ruby'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:93:in `block in compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/ruby.rb:88:in `compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails2.rb:49:in `block in compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails3.rb:38:in `block in compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails3.rb:37:in `compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails4.rb:41:in `block in compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails4.rb:40:in `compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/bin/compile:23:in `block (2 levels) in <main>'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:129:in `log'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/bin/compile:22:in `block in <main>'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:35:in `block in trace'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:35:in `trace'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/bin/compile:18:in `<main>'
 !     undefined method `read' for nil:NilClass
 !
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/rails2.rb:47:in `compile'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:48:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/base.rb:44:in `instrument'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:35:in `call'
	from /tmp/buildpacks/fda2b356ce7de642197a27faa3892370/lib/language_pack/instrument.rb:16:in `instrument'
 !
Failed to compile droplet
Exit status 223
Staging failed: Exited with status 223

FAILED
Error restarting application: BuildpackCompileFailed
```

**Comment by Dannyzen** [comment]
Hey @lnhrdt thanks for putting this issue in. I'm able to reproduce your issue and can confirm we've fixed this bug in the develop branch and plan on releasing the fix to master shortly. You can test this with `cf push appname -b "https://github.com/cloudfoundry/ruby-buildpack#develop"

**Comment by Dannyzen** [comment]
This fix is now in master and available in ruby-buildpack [1.6.20](https://github.com/cloudfoundry/ruby-buildpack/releases/tag/v1.6.20)

**Comment by lnhrdt** [comment]
Thank you @Dannyzen for the fast response and for the fix!
